tests/unit/test_utils.py

```diff
@@ -1,9 +1,236 @@
 import re
 
+import pyam
+import numpy as np
+import pandas as pd
 import pytest
 
-from silicone.utils import _get_unit_of_variable
+from silicone.utils import _get_unit_of_variable, find_matching_scenarios, make_interpolator
 
+_mc = "model_c"
+_sa = "scen_a"
+_sb = "scen_b"
+_eco2 = "Emissions|CO2"
+_gtc = "Gt C/yr"
+_ech4 = "Emissions|CH4"
+_mtch4 = "Mt CH4/yr"
+_msrvu = ["model", "scenario", "region", "variable", "unit"]
+simple_df = pd.DataFrame(
+        [
+            [_mc, _sa, "World", _eco2, _gtc, 0, 1000, 5000],
+            [_mc, _sb, "World", _eco2, _gtc, 1, 1000, 5000],
+            [_mc, _sa, "World", _ech4, _mtch4, 0, 300, 500],
+            [_mc, _sb, "World", _ech4, _mtch4, 1, 300, 500],
+        ],
+        columns=_msrvu + [2010, 2030, 2050],
+    )
+simple_df = pyam.IamDataFrame(simple_df)
+
+
+def test_find_matching_scenarios():
```

Comment:
can you split this into three tests or add some comments or use `pytest.mark.parametrize`? The first assert is testing that you get the right scenario if you put an exact match in? The second is checking that if you give an inexact match you get the right match and the third is checking that if you push the data slightly towards the other scenario you get the right match?

Reply:
ok, uses parametrize now